src/game/collision.cpp

```diff
@@ -202,3 +202,60 @@ void CCollision::MoveBox(vec2 *pInoutPos, vec2 *pInoutVel, vec2 Size, float Elas
 	*pInoutPos = Pos;
 	*pInoutVel = Vel;
 }
+
+bool CCollision::HitTileDeath(vec2& Pos, vec2& Newpos, vec2& Deathpos, float radius)
+{
+    vec2 Dir, Orth, Deathpos1, Deathpos2;
+
+    //calculathe orthogonal vector to move directions
+    Dir = Pos-Newpos;
+    Orth = normalize(vec2(Dir.y, -Dir.x)) * radius;
+
+    //check if hit with deathtile appears with right and left side of tee
+    if(CheckDeath(Pos+Orth, Newpos+Orth, Deathpos1) || CheckDeath(Pos-Orth, Newpos-Orth, Deathpos2))
+    {
+        //choose closer death position
+        if(distance(Pos, Deathpos1) > distance(Pos, Deathpos2))
+            Deathpos = Deathpos2+Orth;
+        else
+            Deathpos = Deathpos1-Orth;
+        return true;
+    }
+    Deathpos = Newpos;
+    return false;
+}
+
+// finds any deathtile between pos1 and pos2 using digital differential analyser
+bool CCollision::CheckDeath(vec2 Pos1, vec2 Pos2, vec2& Deathpos)
+{
+    float dx = (Pos2.x - Pos1.x);
+    float dy = (Pos2.y - Pos1.y);
+
+    float Step;
+    if(fabs(dx) >= fabs(dy))
+        Step = fabs(dx);
+    else
+        Step = fabs(dy);
+
+    dx = dx / Step;
+    dy = dy / Step;
+
+    float x = Pos1.x;
+    float y = Pos1.y;
+
+    int i = 1;
+
+    while(i <= Step)
+    {
+        if(GetCollisionAt(x, y)&COLFLAG_DEATH) {
+            Deathpos = vec2(x, y);
+            return true;
+        }
+        x = x + dx;
+        y = y + dy;
+
+        ++i;
+    }
+    Deathpos = vec2(x, y);
```

Comment:
I don't understand why `Deathpos` should be set here, when no collision with a death tile happened. Imo when only one `CheckDeath` returns true, `HitTileDeath` should not choose the closer death position but simply the only actual death position. Am I missing something?

Reply:
Line 259 is simply there, that Deathpos contains valid information, we could simply delete it

I didn't know you actually had a written code style guide. I will work through that, but the HitTileDeath method should look another way

Reply:
> I didn't know you actually had a written code style guide.

I didn't write any of that, almost all the documentation is written by previous developers, probably matricks. It's not necessarily my favorite code style either, it's just what Teeworlds uses and inherits.

The documentation's github lives there btw: https://github.com/teeworlds/documentation